jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+from astropy.coordinates.builtin_frames import __all__ as all_astropy_frames
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+
+from pyvo.utils import vocabularies
+from pyvo import registry
+from pyvo.dal.exceptions import DALFormatError
+from requests.exceptions import ConnectionError as RequestConnectionError
+from traitlets import Bool, Unicode, Any, List, Float, observe
+
+from jdaviz.configs.imviz.plugins.orientation.orientation import (
+    orientation_plugin_label,
+)
+from jdaviz.core.events import (
+    SnackbarMessage,
+    AddDataMessage,
+    RemoveDataMessage,
+    LinkUpdatedMessage,
+)
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import (
+    PluginTemplateMixin,
+    AddResultsMixin,
+    TableMixin,
+    ViewerSelect,
+    UnitSelectPluginComponent,
+)
+
+__all__ = ["VoPlugin"]
+vo_plugin_label = "Virtual Observatory"
+
+
+@tray_registry("VoPlugin", label=vo_plugin_label)
+class VoPlugin(PluginTemplateMixin, AddResultsMixin, TableMixin):
+    """Plugin to query the Virtual Observatory and load data into Imviz"""
+
+    template_file = __file__, "vo_plugin.vue"
+
+    viewer_items = List([]).tag(sync=True)
+    viewer_selected = Unicode().tag(sync=True)
+
+    wavebands = List().tag(sync=True)
+    waveband_selected = Any().tag(sync=True)  # Any to accept Nonetype
+    resource_filter_coverage = Bool(False).tag(sync=True)
+    resources = List([]).tag(sync=True)
+    resource_selected = Any().tag(sync=True)  # Any to accept Nonetype
+    resources_loading = Bool(False).tag(sync=True)
+
+    source = Unicode("").tag(sync=True)
+    coord_follow_viewer_pan = Bool(False).tag(sync=True)
+    viewer_centered = Bool(False).tag(sync=True)
+    coordframes = List([]).tag(sync=True)
+    coordframe_selected = Unicode("icrs").tag(sync=True)
+    radius_val = Float(1).tag(sync=True)
+    radius_unit_items = List().tag(sync=True)
+    radius_unit_selected = Unicode("deg").tag(sync=True)
+
+    results_loading = Bool(False).tag(sync=True)
+    data_loading = Bool(False).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.viewer = ViewerSelect(
+            self, "viewer_items", "viewer_selected", manual_options=["Manual"]
+        )
+
+        self.radius_unit = UnitSelectPluginComponent(
+            self, items="radius_unit_items", selected="radius_unit_selected"
+        )
+        self.radius_unit.choices = ["deg", "rad", "arcmin", "arcsec"]
+
+        # Waveband properties to filter available registry resources
+        self.wavebands = [
+            w.lower() for w in vocabularies.get_vocabulary("messenger")["terms"]
+        ]
+        self.waveband_selected = None
```

Comment:
can this use a `SelectPluginComponent`?

Reply:
Done in https://github.com/spacetelescope/jdaviz/pull/2872/commits/5d7424f9e7d7bb726fccc6cb92688c2f6a0566b1!